test/e2e/kubevirt.go

```diff
@@ -2521,4 +2563,62 @@ ethernets:
 			)), "second VM should not be ready due to MAC conflict")
 		})
 	})
+
+	Context("subnet exhaustion", func() {
+		const networkName = "net1"
+		var (
+			cudn     *udnv1.ClusterUserDefinedNetwork
+			cidrIPv4 = "10.130.0.0/30" // no usable IPs (.1 and .2, .0 is network infra owned, .3 is broadcast)
+			cidrIPv6 = "2011:100:200::0/120"
+		)
+
+		BeforeEach(func() {
+			l := map[string]string{
+				"e2e-framework":           fr.BaseName,
+				RequiredUDNNamespaceLabel: "",
+			}
+			ns, err := fr.CreateNamespace(context.TODO(), fr.BaseName, l)
+			Expect(err).NotTo(HaveOccurred())
+			fr.Namespace = ns
+			namespace = fr.Namespace.Name
+
+			dualCIDRs := filterDualStackCIDRs(fr.ClientSet, []udnv1.CIDR{udnv1.CIDR(cidrIPv4), udnv1.CIDR(cidrIPv6)})
+			cudn, _ = kubevirt.GenerateCUDN(namespace, networkName, udnv1.NetworkTopologyLayer2, udnv1.NetworkRolePrimary, dualCIDRs)
+			createCUDN(cudn)
+		})
+
+		It("should fail when subnet is exhausted", func() {
+			By("Creating VM that should fail due to subnet exhaustion")
+			exhaustedVMName := "exhausted-vm"
+			vm := fedoraWithTestToolingVM(
+				nil, // labels
+				nil, // no static IP annotations
+				nil, // nodeSelector
+				kubevirtv1.NetworkSource{
+					Pod: &kubevirtv1.PodNetwork{},
+				},
+				`#cloud-config
+password: fedora
+chpasswd: { expire: False }
+`,
+				`version: 2
+ethernets:
+  eth0:
+    dhcp4: true
+    dhcp6: true
+    ipv6-address-generation: eui64`,
```

Comment:
isn't this the default ?

Reply:
I don't know really, this is [taken](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/d93369dc7627667e578d8c7f6cbe7cfb5af13f5c/test/e2e/kubevirt.go#L2402) from the functions above (createVMWithStaticIP).

Reply:
Nope, the latest NetworkManager default is generate the IPv6 LLA at random and ignore the mac addres, this do not work with ovn-kubernetes port security things.